node/src/main.rs

```diff
@@ -866,70 +880,187 @@ async fn main() {
 }
 
 /// Parses an Ethereum connection string and returns the network name and Ethereum adapter.
-fn parse_ethereum_networks_and_nodes(
+fn parse_ethereum_networks(
     logger: Logger,
     networks: clap::Values,
     connection_type: ConnectionType,
     registry: Arc<MetricsRegistry>,
-) -> Result<HashMap<String, Arc<dyn EthereumAdapterTrait>>, Error> {
+) -> Result<EthereumNetworks, Error> {
     let eth_rpc_metrics = Arc::new(ProviderEthRpcMetrics::new(registry));
-    networks
-        .map(|network| {
-            if network.starts_with("wss://")
-                || network.starts_with("http://")
-                || network.starts_with("https://")
-            {
-                return Err(format_err!(
-                    "Is your Ethereum node string missing a network name? \
+    networks.fold(Ok(EthereumNetworks::new()), |networks, network_arg| {
+        if network_arg.starts_with("wss://")
+            || network_arg.starts_with("http://")
+            || network_arg.starts_with("https://")
+        {
+            return Err(format_err!(
+                "Is your Ethereum node string missing a network name? \
                      Try 'mainnet:' + the Ethereum node URL."
-                ));
-            } else {
-                // Parse string (format is "NETWORK_NAME:URL")
-                let split_at = network.find(':').ok_or_else(|| {
-                    return format_err!(
-                        "A network name must be provided alongside the \
+            ));
+        } else {
+            // Parse string (format is "NETWORK_NAME:NETWORK_CAPABILITIES:URL" OR
+            // "NETWORK_NAME::URL" which will default to NETWORK_CAPABILITIES="archive,traces")
+            let split_at = network_arg.find(':').ok_or_else(|| {
+                return format_err!(
+                    "A network name must be provided alongside the \
                          Ethereum node location. Try e.g. 'mainnet:URL'."
-                    );
-                })?;
+                );
+            })?;
 
-                let (name, loc_with_delim) = network.split_at(split_at);
-                let loc = &loc_with_delim[1..];
+            let (name, rest_with_delim) = network_arg.split_at(split_at);
+            let rest = &rest_with_delim[1..];
+            if name.is_empty() {
+                return Err(format_err!(
+                    "Ethereum network name cannot be an empty string"
+                ));
+            }
 
-                if name.is_empty() {
-                    return Err(format_err!(
-                        "Ethereum network name cannot be an empty string"
-                    ));
-                }
+            let http_split_at = rest.find("http").ok_or_else(|| {
```

Comment:
I don't think you want to split at `http` or so, but at the next `:`. That seems more robust.